### ISSUE TITLE
v8.1.0: "Couldn't find preset "node5" relative to directory"

### ISSUE THREAD

**Issue by atkinchris** [AtomLinter/linter-eslint#796]
## Issue Type
Bug introduced with version `8.1.0`.

## Issue Description
When using [`eslint-import-resolver-webpack`](https://github.com/benmosher/eslint-plugin-import/tree/master/resolvers/webpack) with a `webpack.config.babel.js`, `linter-eslint` no longer lints files in the project.

The below console output is observed in Atom, when no linting occurs:

```
[Linter-ESLint] STDERR There was a problem loading formatter: /Users/christopher.atkin/.atom/packages/linter-eslint/lib/reporter.js
Error: Couldn't find preset "node5" relative to directory "/Users/christopher.atkin/.atom/packages/linter-eslint"
```

Reverting to `linter-eslint@8.0.0` (or using a non-Babel `webpack.config.js`) resolved the issue.

## Bug Checklist
Atom restarted, freshly reinstalled all plugins and Atom itself, and moved between `linter-eslint` versions to confirm.

```
Atom version: 1.13.1
linter-eslint version: 8.1.0
ESLint version: 3.14.1
Hours since last Atom restart: 0.1
Platform: darwin
Using local project ESLint from /Users/christopher.atkin/Source/promcomm-app/node_modules/eslint
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": []
}
```

**Comment by atkinchris** [comment]
Installing `babel-preset-node5` as a non-dev dependency into `~/.atom/packages/linter-eslint` also resolves this. Could this have been introduced following https://github.com/AtomLinter/linter-eslint/commit/5b4f68291518a04c181bba1c7c8bdd0090e1f2fa?

**Comment by atkinchris** [comment]
Looking further into this, I suspect this comes from the `'use babel'` pragma mark in each file (https://github.com/AtomLinter/linter-eslint/blob/master/lib/helpers.js#L2) causing Atom to preprocess package files (ref: http://blog.atom.io/2015/02/04/built-in-6to5.html).

By changing to a `.babelrc` file in https://github.com/AtomLinter/linter-eslint/commit/5b4f68291518a04c181bba1c7c8bdd0090e1f2fa, Babel uses this to determine which `presets` to use. As `node5` is not installed as a regular dependency, it will not be available to Atom when it attempts to preprocess the `eslint` files that need transpiling.

**Comment by Arcanemagus** [comment]
This sounds like a bug in `eslint-import-resolver-webpack`, the `linter-eslint` code should have _nothing_ to do with linting your project, especially since your project is properly loading a local ESLint instance. I have no idea how it's even getting a reference to it 😕.

Can you provide a minimal repro case?

**Comment by atkinchris** [comment]
I agree, @Arcanemagus - it feels like it should be a bug in `eslint-import-resolver-webpack`, or at the very least, an incompatibility. The version bump in this project surfaced the error, and gave the related error message, hence my starting here.

I'll try to reproduce a minimal case.

**Comment by atkinchris** [comment]
I cannot create a succinct reproduction for this issue, that doesn't require a range of ESLint plugins (and that doesn't return a different error, handled gracefully through `linter-eslint`).

Happy to close this as a compatibility issue, when mixing a specific set of plugins with an edge usage case.

**Comment by beauroberts** [comment]
Hey, I also ran into this issue because I'm sure I have the same weird combination of eslint plugins installed. Is the current recommendation to downgrade `linter-eslint`? Has an issue been filed against `eslint-import-resolver-webpack`?

**Comment by nicolasartman** [comment]
I was able to solve this by pulling my `resolve` configuration out into a new es5-only file and exporting it using commonjs form.

For example:

__webpack-resolve-configuration.js__
```
module.exports = {
  resolve: {
    root: 'src',
    extensions: ['', '.js', '.jsx', '.json'],
    modulesDirectories: ['node_modules/']
  }
};
```

Then I just import that file into my main webpack configuration file to use for the `resolve` configuration field, and also reference it with my eslint config for the import plugin:

```
  "settings": {
    "import/resolver": {
      "webpack": {
        "config": "webpack-resolve-configuration.js"
      }
    },
  }
```

It's not a perfect solution but it's simple, it works, and you don't have to duplicate code or downgrade your whole webpack config to es5.

**Comment by Arcanemagus** [comment]
Can either of you determine a sharable reproduction case? Has anyone filed this bug with `eslint-import-resolver-webpack`?

**Comment by nicolasartman** [comment]
It took a bit but I think I have it pretty well narrowed down. Here's a reproduction—in v8.0.0 it lints fine, but in v8.1.0 it silently fails with the error.

[repro.zip](https://github.com/AtomLinter/linter-eslint/files/757298/repro.zip)

To fix it in v8.1 just change the config to reference babel.config.js, delete the import in that file, and change the export to module.exports format.

The problem seems to be that the webpack config, when referenced through the resolver, doesn't get transpiled. I haven't been able to investigate exactly why, but I hope this reproduction helps!

[repro-fixed.zip](https://github.com/AtomLinter/linter-eslint/files/757373/repro-fixed.zip)

**Comment by SpainTrain** [comment]
Deeper explanation in https://github.com/AtomLinter/linter-eslint/pull/809, but TLDR: this is caused by how `babel/register` finds and processes `.babelrc` files.

`.babelrc` appears to only be used at build-time, so isn't needed in the installed APM package.  Reopening this issue because:

- not an issue specific to `eslint-import-resolver-webpack`
- arguably a `linter-eslint` bug given that a config file in the package artifact references a module that is not a production dep
- the fix is all ready  #806 :smile_cat:
